### PR TITLE
ACS-3964: fix for non-inheritable rules still appearing in inherited rule sets

### PR DESCRIPTION
**Pull request by gevagel**
This PR introduces changes to avoid non-inheritable rules appearing in inherited rule sets. This is the fix to the following JIRA issue [https://alfresco.atlassian.net/browse/ACS-3964](https://alfresco.atlassian.net/browse/ACS-3964)

### CODE REVIEW

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/GetInheritedRulesTests.java [hunk 1]
```diff
@@ -91,6 +96,37 @@ public void getInheritedRules()
                       .and().entriesListCountIs(1);
     }
 
+    /**
+     * Check we only get the owned rules and no inherited rules in the child folder when parent rule isn't inheritable.
+     */
+    @Test (groups = { TestGroup.REST_API, TestGroup.RULES, TestGroup.SANITY })
```
Comment (KikoUK) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/GetInheritedRulesTests.java:102]: Is this e2e correct for what's implemented? I think you should create a rule in the parent folder which has inherited set as false and then ensure that the rule is not inherited by the child. 

Also, why are you creating a rule in the child folder?

File: remote-api/src/test/java/org/alfresco/rest/api/impl/rules/RulesImplTest.java [hunk 2]
```diff
@@ -141,6 +155,68 @@ public void testGetRules()
...
+        assertThat(rulesPage.getCollection().stream().findFirst().get()).isEqualTo(ruleMock);
+    }
+
+    @Test
+    public void testGetRules_rule_not_applied_to_children()
+    {
+        given(ruleSetMock.getInclusionType()).willReturn(InclusionType.INHERITED);
+        ruleModelInherited.applyToChildren(false);
+
+        //given(ruleSetMock.getInclusionType()).willReturn(InclusionType.OWNED);
```
Comment (krdabrowski) [inline remote-api/src/test/java/org/alfresco/rest/api/impl/rules/RulesImplTest.java:179]: If this check is not necessary here, maybe it should be entirely removed?

**Review by krdabrowski — APPROVED** [review]

**Review by mpichura — COMMENTED** [review]
Most of my remarks are minor and all of them consider test classes.

**Comment by gevagel** [comment]
Thanks for the reviews. Comments have been addressed and PR should look better now

**Review by mpichura — APPROVED** [review]
LGTM now

**Review by KikoUK — APPROVED** [review]

**Pull request merged**